### PR TITLE
feat: add company name

### PR DESCRIPTION
**Pull request by PritishBudhiraja**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/HyperSwitch/Others/CompanyNameModal.res [hunk 1]
```diff
@@ -0,0 +1,89 @@
+@react.component
+let make = (~showModal, ~setShowModal) => {
+  open APIUtils
+  open LogicUtils
+  let showToast = ToastState.useShowToast()
+  let updateDetails = useUpdateMethod(~showErrorToast=false, ())
+  let merchantId = HSLocalStorage.getFromMerchantDetails("merchant_id")
+  let setMerchantDetailsValue = Recoil.useSetRecoilState(HyperswitchAtom.merchantDetailsValueAtom)
```
Comment (Riddhiagrawal001) [inline src/screens/HyperSwitch/Others/CompanyNameModal.res:8]: use `->` operator

File: src/screens/HyperSwitch/Others/CompanyNameModal.res [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    <Form
+      key="merchant_name-validation"
+      initialValues={Dict.make()->JSON.Encode.object}
+      onSubmit
+      validate={values => values->validateForm}>
+      <div className="flex flex-col gap-6 h-full w-full">
+        <FormRenderer.DesktopRow>
+          <FormRenderer.FieldRenderer
+            fieldWrapperClass="w-full"
+            field={FormRenderer.makeFieldInfo(
```
Comment (Riddhiagrawal001) [inline src/screens/HyperSwitch/Others/CompanyNameModal.res:70]: assign FormRenderer.makeFieldInfo  to a variable and replace the variable here fir better readability

**Review by JeevaRamu0104 — COMMENTED** [review]
check for merchantRead access

File: src/screens/HyperSwitch/SwitchMerchant/SwitchMerchant.res [hunk 3]
```diff
@@ -55,49 +55,44 @@ module NewAccountCreationModal = {
...
-                  <FormRenderer.SubmitButton text="Create Account" buttonSize={Small} />
-                </div>
+        </div>
+        <Form key="new-account-creation" onSubmit>
+          <div className="flex flex-col gap-12 h-full w-full">
+            <FormRenderer.DesktopRow>
+              <div className="flex flex-col gap-5">
+                <FormRenderer.FieldRenderer
+                  fieldWrapperClass="w-full"
+                  field={FormRenderer.makeFieldInfo(
```
Comment (Riddhiagrawal001) [inline src/screens/HyperSwitch/SwitchMerchant/SwitchMerchant.res:77]: Same here. Assign the FormRenderer.makeFieldInfo to a variable and replace with variable here

File: src/screens/HyperSwitch/Others/CompanyNameModal.res [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    (),
+  )
+
+  <Modal
+    showModal
+    paddingClass=""
+    modalHeading="Welcome aboard! Let's get started"
+    setShowModal
+    showCloseIcon=false
+    modalHeadingDescription="Start by creating your business profile"
```
Comment (Riddhiagrawal001) [inline src/screens/HyperSwitch/Others/CompanyNameModal.res:65]: Change it to a better text for creating a company name

Comment (PritishBudhiraja) [reply src/screens/HyperSwitch/Others/CompanyNameModal.res]: Updated

**Review by Riddhiagrawal001 — APPROVED** [review]

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**